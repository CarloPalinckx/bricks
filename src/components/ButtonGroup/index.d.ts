import { SFC } from 'react';
declare const ButtonGroup: SFC;
export default ButtonGroup;
