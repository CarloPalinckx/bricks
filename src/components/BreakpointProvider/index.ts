export { default } from './BreakpointProvider.template';
