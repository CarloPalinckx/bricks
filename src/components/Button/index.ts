export { default as default, ButtonThemeType } from './Button.style';
