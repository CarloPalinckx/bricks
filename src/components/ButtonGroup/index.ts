export { default as default } from './ButtonGroup.template';
