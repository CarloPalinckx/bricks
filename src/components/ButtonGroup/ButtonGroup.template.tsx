import React, { Children, StatelessComponent } from 'react';
import trbl from '../../utility/trbl';
import Box from '../Box';
import BreakpointProvider from '../BreakpointProvider';

const ButtonGroup:StatelessComponent = (props):JSX.Element => {
    return (
        <BreakpointProvider
            breakpoints={{
                small: 0,
                medium: 0,
                large: 375,
            }}
        >
            {(breakpoint):JSX.Element => {
                const direction = breakpoint === 'small' ? 'column' : 'row-reverse';
                const alignItems = breakpoint === 'small' ? 'stretch' : 'center';

                return (
                    <Box
                        direction={direction}
                        justifyContent="flex-start"
                        alignItems={alignItems}
                        wrap={true}
                        margin={trbl(-6)}
                    >
                        {Children.map(props.children, (child):JSX.Element => (
                            <Box direction="column" justifyContent="stretch" margin={trbl(6)}>
                                {child}
                            </Box>
                        ))}
                    </Box>
                );
            }}
        </BreakpointProvider>
    );
};

export default ButtonGroup;
