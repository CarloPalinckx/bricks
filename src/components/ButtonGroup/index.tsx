import React, { Children, StatelessComponent } from 'react';
import trbl from '../../utility/trbl';
import Box from '../Box';
import BreakpointProvider from '../BreakpointProvider';

const ButtonGroup: StatelessComponent = (props): JSX.Element => {
    return (
        <BreakpointProvider
            breakpoints={{
                small: 0,
                medium: 0,
                large: 375,
            }}
        >
            {(breakpoint): JSX.Element => {
                const direction =
                    breakpoint === 'small' ? 'column' : 'row-reverse';

                return (
                    <Box
                        direction={direction}
                        justifyContent="flex-start"
                        alignItems="stretch"
                        wrap
                        margin={trbl(-6)}
                    >
                        {Children.map(props.children, (child): JSX.Element => (
                            <Box
                                direction={
                                    breakpoint === 'small' ? 'column' : 'row'
                                }
                                alignSelf="stretch"
                                margin={trbl(6)}
                            >
                                {child}
                            </Box>
                        ))}
                    </Box>
                );
            }}
        </BreakpointProvider>
    );
};

export default ButtonGroup;
