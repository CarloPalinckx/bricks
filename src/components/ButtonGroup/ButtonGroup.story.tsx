import { storiesOf } from '@storybook/react';
import React from 'react';
import styled from 'styled-components';
import ButtonGroup from '.';
import Button from '../Button';
import FlatButton from '../FlatButton';

const Wrap = styled.div`
    max-width: 640px;
    border: solid 3px rgba(255, 36, 94, 0.3);
`;

storiesOf('ButtonGroup', module)
    .add('Default', () => {
        return (
            <Wrap>
                <ButtonGroup>
                    <Button title="Primary button">Primary button</Button>
                    <FlatButton title="Primary button">Cancel button</FlatButton>
                </ButtonGroup>
            </Wrap>
        );
    });
