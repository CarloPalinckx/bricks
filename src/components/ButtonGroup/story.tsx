import { storiesOf } from '@storybook/react';
import React from 'react';
import styled from 'styled-components';
import ButtonGroup from '.';
import Button from '../Button';

const Wrap = styled.div`
    max-width: 640px;
    border: solid 3px rgba(255, 36, 94, 0.3);
`;

storiesOf('ButtonGroup', module).add('Default', () => {
    return (
        <Wrap>
            <ButtonGroup>
                <Button variant="primary" title="Primary button">
                    Primary button
                </Button>
                <Button variant="secondary" title="Secondary button">
                    Secondary button
                </Button>
            </ButtonGroup>
        </Wrap>
    );
});
