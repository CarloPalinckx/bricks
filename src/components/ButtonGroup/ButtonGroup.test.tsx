import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import React from 'react';
import ButtonGroup from '.';
import BreakpointProvider from '../BreakpointProvider';
import { PropsType } from '../BreakpointProvider/BreakpointProvider.template';
import Button from '../Button';
import FlatButton from '../FlatButton';

jest.mock('../BreakpointProvider', () => jest.fn().mockImplementation((props:PropsType):JSX.Element => {
    return props.children('large');
}));

describe('ButtonGroup', () => {
    it('renders with a Button and a FlatButton in a large node', () => {
        const component = shallow(
            <ButtonGroup>
                <Button title="Test" variant="primary" />
                <FlatButton title="Test" />
            </ButtonGroup>
        ).dive();

        expect(toJson(component)).toMatchSnapshot();
    });

    it('renders with a single Button in a small node', () => {
        (BreakpointProvider as jest.Mock<BreakpointProvider>).mockImplementationOnce((props:PropsType):JSX.Element => {
            return props.children('small');
        });

        const component = shallow(
            <ButtonGroup>
                <Button title="Test" variant="primary" />
            </ButtonGroup>
        ).dive();

        expect(toJson(component)).toMatchSnapshot();
    });
});
