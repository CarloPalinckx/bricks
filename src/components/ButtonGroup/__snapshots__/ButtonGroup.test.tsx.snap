// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ButtonGroup renders with a Button and a FlatButton in a large node 1`] = `
<Styled(Box)
  alignItems="center"
  direction="row-reverse"
  justifyContent="flex-start"
  margin={
    Object {
      "bottom": "-6px",
      "left": "-6px",
      "right": "-6px",
      "top": "-6px",
    }
  }
  wrap={true}
>
  <Styled(Box)
    direction="column"
    justifyContent="stretch"
    key=".0"
    margin={
      Object {
        "bottom": "6px",
        "left": "6px",
        "right": "6px",
        "top": "6px",
      }
    }
  >
    <Styled(Button)
      title="Test"
      variant="primary"
    />
  </Styled(Box)>
  <Styled(Box)
    direction="column"
    justifyContent="stretch"
    key=".1"
    margin={
      Object {
        "bottom": "6px",
        "left": "6px",
        "right": "6px",
        "top": "6px",
      }
    }
  >
    <Styled(FlatButton)
      title="Test"
    />
  </Styled(Box)>
</Styled(Box)>
`;

exports[`ButtonGroup renders with a single Button in a small node 1`] = `
<Styled(Box)
  alignItems="stretch"
  direction="column"
  justifyContent="flex-start"
  margin={
    Object {
      "bottom": "-6px",
      "left": "-6px",
      "right": "-6px",
      "top": "-6px",
    }
  }
  wrap={true}
>
  <Styled(Box)
    direction="column"
    justifyContent="stretch"
    key=".0"
    margin={
      Object {
        "bottom": "6px",
        "left": "6px",
        "right": "6px",
        "top": "6px",
      }
    }
  >
    <Styled(Button)
      title="Test"
      variant="primary"
    />
  </Styled(Box)>
</Styled(Box)>
`;
